src/libraries/Common/src/System/Net/IPv4AddressHelper.Common.cs

```diff
@@ -14,10 +14,24 @@ internal static partial class IPv4AddressHelper
         private const int NumberOfLabels = 4;
 
         // Only called from the IPv6Helper, only parse the canonical format
-        internal static unsafe int ParseHostNumber(ReadOnlySpan<char> str, int start, int end)
+        internal static int ParseHostNumber(ReadOnlySpan<char> str, int start, int end)
         {
-            byte* numbers = stackalloc byte[NumberOfLabels];
-            ParseCanonical(str, numbers, start, end);
+            Span<byte> numbers = stackalloc byte[NumberOfLabels];
+
+            for (int i = 0; i < numbers.Length; ++i)
+            {
+                int b = 0;
+                char ch;
+
+                for (; (start < end) && (ch = str[start]) != '.' && ch != ':'; ++start)
+                {
+                    b = (b * 10) + ch - '0';
+                }
+
+                numbers[i] = (byte)b;
+                ++start;
+            }
+
```

Comment:
Another place where I removed the unsafe code in favor of using spans.  Perf is unchanged, with the JIT eliminating the bounds checks on the numbers span.  This was the sole call site to ParseCanonical, so I just inlined it.